core/include/core/detector.hpp

```diff
@@ -152,38 +227,51 @@ namespace detray
         using discs = dvector<disc_mask>;
         using surface_links = bool;
 
-        using surface_type_map = dmap<guaranteed_index, guaranteed_index>;
-        using detector_surface = surface<guaranteed_index, typed_guaranteed_range, guaranteed_index>;
+        using surface_type_map = dmap<dindex, dindex>;
+        using detector_surface = surface<dindex, typed_dindex_range, dindex>;
 
         /** Method to add a list of portals to the portal tuple
          * 
-         * @tparam portal_type the type of the portal mask
+         * @tparam mask_type the type of the portal mask
          * 
+         * @param volume_index the volume index to which these surfaces should be added
          * @param transform the transform of the portal surface
          * @param portals the vector of portal masks
-         * @param volume to which this portal belongs to
+         * 
          **/
         template <typename mask_type>
-        void add_surfaces(dvector<transform_type> transforms, const typename mask_type::mask_values &mask_values, volume &volume)
+        void add_surfaces(dindex volume_index, dvector<transform_type> transforms, const typename mask_type::mask_values &mask_values)
         {
+            auto &volume = _volumes[volume_index];
             // Get the boundary group, record the index and insert the portals
             auto &mask_group = std::get<dvector<mask_type>>(_surface_masks);
-            guaranteed_index mask_index = mask_group.size();
+            dindex mask_index = mask_group.size();
             mask_type mask;
             mask = mask_values;
             mask_group.push_back(std::move(mask));
-            guaranteed_index type = _surface_types.find(mask_type::mask_identifier)->second;
-            typed_guaranteed_range typed_mask_range = {type, {mask_index, mask_index}};
-            guaranteed_index volume_index = volume.index;
+            dindex type = _surface_types.find(mask_type::mask_identifier)->second;
+            typed_dindex_range typed_mask_range = {type, {mask_index, mask_index}};
             for (auto transform : transforms)
             {
                 volume.surface_indices.push_back(_surfaces.size());
-                guaranteed_index transform_index = _surface_transforms.size();
+                dindex transform_index = _surface_transforms.size();
                 _surface_transforms.push_back(std::move(transform));
                 _surfaces.push_back(detector_surface(std::move(transform_index), std::move(typed_mask_range), std::move(volume_index), false));
             }
         }
 
+        /** Add surface finders to a volume
+         *
+         * @param volume_index the volume index to which these finders should be added
+         * @param surface_finders the local finders that are to be added to this volume
+         */
+         dindex_range add_surface_finders(dindex volume_index, dvector<local_object_finder> surface_finders ){
+            auto &volume = _volumes[volume_index];
```

Comment:
Why pass the volume if it is never connected to the finders?

Reply:
good question, I think that was just a left-over.

Reply:
Deleted it.